### PR TITLE
Added analytics service J=PC-23360

### PR DESCRIPTION
**Pull request by tbirdsall**

### CODE REVIEW

File: analytics_data.go [hunk 1]
```diff
@@ -0,0 +1,412 @@
+package yext
+
+type AnalyticsData struct {
+	ProfileViews                          *int     `json:"Profile Views"`
```
Comment (mhupman) [inline analytics_data.go]: do we need the JSON tags?  They seem funky

File: analytics_service.go [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+	FoursquareCheckinGender    *string   `json:"foursquareCheckinGender"`
+	FoursquareCheckinTimeOfDay *string   `json:"foursquareCheckinTimeOfDay"`
+	InstagramContentType       *string   `json:"instagramContentType"`
+	Age                        *[]string `json:"age"`
+	Gender                     *string   `json:"gender"`
+	FacebookImpressionType     *[]string `json:"facebookImpressionType"`
+	FacebookStoryType          *[]string `json:"facebookStoryType"`
+}
+
+type AnalyticsReportBody struct {
```
Comment (mhupman) [inline analytics_service.go]: AnalyticsReportRequest?

File: analytics_service.go [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+	Dimensions []string          `json:"dimensions"`
+	Filters    *AnalyticsFilters `json:"filters"`
+}
+
+type AnalyticsReportResponse struct {
+	Data []*AnalyticsData `json:"data"`
+	Id   int              `json:"id"`
+}
+
+func (l *AnalyticsService) Create(jsonData *AnalyticsReportBody) (*AnalyticsReportResponse, *Response, error) {
```
Comment (mhupman) [inline analytics_service.go]: prefer `a`, `arr`, `req` as argument name.  its not actually JSON.

File: analytics_data.go (hunk 1 above)
Comment (tbirdsall) [reply analytics_data.go]: Hmm, not quite sure - I was not able to get the api call to work without it. What do you mean by "funky"?

Comment (mhupman) [reply analytics_data.go]: They have spaces in them

Comment (tbirdsall) [reply analytics_data.go]: Yep! That's purposeful, that's how they are in the response (verified in Postman and dev logs). Interestingly, all the report data based on metrics are uppercase with spaces in between, but the ones based on dimensions are lowercase and camelcased

Comment (mhupman) [reply analytics_data.go]: Interesting, cool!

**Pull request closed without merging**